backend/api/task.proto

```diff
@@ -0,0 +1,110 @@
+// Copyright 2021 The Kubeflow Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+syntax = "proto3";
+
+option go_package = "github.com/kubeflow/pipelines/backend/api/go_client";
+package api;
+
+import "google/protobuf/empty.proto";
+import "google/api/annotations.proto";
+import "google/protobuf/timestamp.proto";
+import "backend/api/resource_reference.proto";
+import "protoc-gen-swagger/options/annotations.proto";
+
+message Task {
+  // Output. Unique run ID. Generated by API server.
+  string id = 1;
+
+  // Required input field. The Namespace to which this pipeline task belongs.
+  string namespace = 2;
+
+  // Required input field. The PipelineName to which this pipeline task belongs.
+  string pipelineName = 3;
```

Comment:
This field can be a little controversial.

Several facts:
* pipeline name is not a unique identifier, `pipeline name + namespace` is (also, namespace can be empty -- means the pipeline is shared by all namespaces). Right now, KFP backend supports namespaced pipelines, but KFP UI didn't integrate the support. So most users should still be only using shared pipelines.
* all other refs use id, only pipeline is using name, this is a minor difference.

I'm struggling with this choice and do not have a preference at the moment. How do you perceive these problems?

Reply:
at least I think it helps if you clarify pipelineName can be used to find the context name in MLMD.

Hmm, so this means that two pipelines in different namespaces may share cache when they have the same name. We might need to think about a solution for this as well.

Reply:
The `pipelineName` means the pipeline context name in MLMD. `pipelineName`  is used as cache scope and it is same with the behavior of TFX. But TFX has a restriction that `pipelineName` need to be unique("pipeline-name: The name of this pipeline. The pipeline name must be unique.
TFX uses the pipeline name when querying ML Metadata for component input artifacts. Reusing a pipeline name may result in unexpected behaviors." FYI: https://www.tensorflow.org/tfx/guide/build_tfx_pipeline#using_the_pipeline_class). I think maybe we should add restriction to make the `pipelineName` unique?

Reply:
Encode namespace in pipeline context name

`namespace/my-namespace/pipeline/my-pipeline` for namespaced pipelines
`pipeline/my-pipeline` for shared pipelines

Reply:
nit: shall we put the format convention in the comments, that feels like an important API contract

Reply:
done